src/transformers/models/audio_spectrogram_transformer/feature_extraction_audio_spectrogram_transformer.py

```diff
@@ -19,12 +19,18 @@
 from typing import List, Optional, Union
 
 import numpy as np
-import torch
-import torchaudio.compliance.kaldi as ta_kaldi
 
+from ...audio_utils import mel_filter_bank, spectrogram, window_function
 from ...feature_extraction_sequence_utils import SequenceFeatureExtractor
 from ...feature_extraction_utils import BatchFeature
-from ...utils import TensorType, logging
+from ...utils import TensorType, is_speech_available, is_torch_available, logging
+
+
+if is_speech_available():
+    import torchaudio.compliance.kaldi as ta_kaldi
+
+if is_torch_available():
```

Comment:
In speech-to-text we bundle these imports into one:
```python
if is_speech_available():
    import torchaudio.compliance.kaldi as ta_kaldi
    import torch
```
Should we do the same here since we can only use `torch` if torchaudio is available?

Reply:
Actually, `torch` is also used here even when `torchaudio` isn't used. I can maybe refactor the code to change that, but I'm not sure it's worth the time, WDYT ?